### DISCUSSION TITLE
CLOG 2.3 - Dev Journal

### DISCUSSION THREAD

**Discussion by rabbibotton** [rabbibotton/clog#364] [General]
Goals:

Faster and more powerful GUI Builder and new Layout Controls for the CLOG Framework - packing boxes, split windows, etc

**Comment by rabbibotton** [comment]
Better garbage collection of elements destroyed in DOM not using CLOG on Lisp side

**Reply by rabbibotton** [reply]
clog-connection:*browser-gc-on-ping* if set to true will auto browser garbage collect in background.

**Comment by rabbibotton** [comment]
Bulk DOM traversal list-all-children
Speed up of builder 400% on Linux

**Reply by rabbibotton** [reply]
I also corrected numerous bugs and enhanced performance even more :)

**Comment by aykaramba** [comment]
Amazing.

**Comment by rabbibotton** [comment]
Speed issues under WSL2 with Linux solved and a fix for WSL2 here - https://www.reddit.com/r/Common_Lisp/comments/1dojy40/wsl2_fix_insanely_fast_clog_builder_and_lisp_web/

.wslconfig
```

[wsl2]

networkingMode=mirrored

```

**Comment by rabbibotton** [comment]
REPL handles multiple results, better backtraces and print outs of results use ~S matching slime

**Comment by rabbibotton** [comment]
Improvements in websocket connectivity to handle longer network connect time in WSL for ARM
Additional APIs for grid layouts bound
ENVELOPE-PANEL - added used to envelope a CLOG Builder Panel or other absolute positioned controls in FLEX and GRIDs

**Reply by rabbibotton** [reply]
mirrored network requires a 7 second delay on intel and 3 on ARM, so it is configurable but by default waits 7 seconds from a disconnect in hopes a connection retries.

**Comment by rabbibotton** [comment]
grid control add to builder

**Comment by rabbibotton** [comment]
You can now change *static-route* at run-time and change where static files are served from.
When you click the run button for a project in builder switches the static route to the projects (even if hit cancel as well)

**Comment by rabbibotton** [comment]
clog-connection:add-plugin-path - use alternate static-paths based on regex - this allows plugins and libraries to share your html directory with its own resources

**Comment by rabbibotton** [comment]
CLOG-ACE and CLOG-TERMINAL updated to use add-plugin-path make sure to pull if using git
Removed the static-files from clog for both since no longer need to have in multiple places

**Comment by rabbibotton** [comment]
Added some pretty printing for html and .clog files and column size limits to help when needing to edit directly

**Comment by rabbibotton** [comment]
destroy-children added - use instead of (setf (text obj) "") or inner-html for destroying all contained html as this will run browser-gc also.

New directory files search - right click on a file icon and choose search and have a GREP experience :)

<img width="1247" alt="image" src="https://github.com/rabbibotton/clog/assets/11616789/0e694671-042d-477a-b506-4c6af8766fc3">

**Comment by rabbibotton** [comment]
Now recursive regex grep

<img width="1244" alt="image" src="https://github.com/rabbibotton/clog/assets/11616789/a56f4087-3865-4a18-b089-309a7d2a68ac">

**Comment by rabbibotton** [comment]
<img width="859" alt="image" src="https://github.com/rabbibotton/clog/assets/11616789/003be455-acdb-49fe-84d7-2644b44ed780">

preview panel

**Comment by rabbibotton** [comment]
Now you have two searches for symbols - Alt-. system browse (like before) and Alt-, regex browse (game changer!)

**Comment by rabbibotton** [comment]
CLOG now supports non-quicklisp based development completely based off of ASDF. This allows for OCICL based systems see https://github.com/rabbibotton/clog/blob/main/OCICL.md

**Comment by rabbibotton** [comment]
CLOG directions also for QLOT -  https://github.com/rabbibotton/clog/blob/main/QLOT.md

**Comment by rabbibotton** [comment]
QuickLisp project directories emulated when using OCICL.
When starting builder using :project (or an OCICL directory) the preferences.lisp file is stored in the projects directory.
Additional projects can be created in the OCICL directory.

**Comment by rabbibotton** [comment]
Better code completion

**Comment by rabbibotton** [comment]
fixed documentation for all packages along with the status bar function parameters
